### PR TITLE
rec: use native QType in lua scripting

### PR DESCRIPTION
**Pull request by mind04**
### Short description
qtype.getName() and qtype.getCode() is now possible without type casting.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/lua-scripting/dq.rst [hunk 1]
```diff
@@ -17,7 +17,11 @@ The DNSQuestion object contains at least the following fields:
 
   .. attribute:: DNSQuestion.qtype
 
-      Type this query is for as an integer, can be compared against ``pdns.A``, ``pdns.AAAA``.
+     .. versionchanged:: 4.2.0
+
+       The qtype parameter is now a QType instead of an int
+
+      Type this query is for as an QType, can be compared against ``pdns.A``, ``pdns.AAAA``.
```
Comment (Habbie) [inline pdns/recursordist/docs/lua-scripting/dq.rst]: This sentence is unreadable (it already was), but could you perhaps fix it for us anyway? :)

**Comment by rgacogne** [comment]
I'm not sure what the impact of this PR is, performance-wise, since we now have to go back to `C++` from `Lua` to do anything with the `qtype`. It might also be a bit annoying that existing scripts will break after an upgrade, but I guess we can deal with that.

**Comment by Habbie** [comment]
> I'm not sure what the impact of this PR is, performance-wise, since we now have to go back to C++ from Lua to do anything with the qtype. It might also be a bit annoying that existing scripts will break after an upgrade, but I guess we can deal with that.

Under LuaJIT, the performance hit of going back to C++ is huge.

**Comment by pieterlexis** [comment]
> It might also be a bit annoying that existing scripts will break after an upgrade, but I guess we can deal with that.

This needs upgrade notes

**Pull request closed without merging**